modules/sharethroughBidAdapter.js

```diff
@@ -138,175 +193,77 @@ export const sharethroughAdapterSpec = {
-    return size[0] * size[1];
+function getVideoApi({ api }) {
+  let defaultValue = [2];
+  if (api && Array.isArray(api) && api.length > 0) {
+    return api;
+  } else {
+    return defaultValue;
   }
-
-  return sizes.reduce((prev, current) => {
-    if (area(current) > area(prev)) {
-      return current
-    } else {
-      return prev
-    }
-  });
 }
 
-function generateAd(body, req) {
-  const strRespId = `str_response_${req.data.bidId}`;
-
-  let adMarkup = `
-    <div data-str-native-key="${req.data.placement_key}" data-stx-response-name="${strRespId}">
-    </div>
-    <script>var ${strRespId} = "${b64EncodeUnicode(JSON.stringify(body))}"</script>
-  `;
-
-  if (req.strData.skipIframeBusting) {
-    // Don't break out of iframe
-    adMarkup = adMarkup + `<script src="https://native.sharethrough.com/assets/sfp.js"></script>`;
+function getVideoProtocols({ protocols }) {
+  let defaultValue = [2, 3, 5, 6, 7, 8];
+  if (protocols && Array.isArray(protocols) && protocols.length > 0) {
+    return protocols;
   } else {
-    // Add logic to the markup that detects whether or not in top level document is accessible
-    // this logic will deploy sfp.js and/or iframe buster script(s) as appropriate
-    adMarkup = adMarkup + `
-      <script>
-        (${sharethroughInternal.isLockedInFrame.toString()})()
-      </script>
-      <script>
-        (${sharethroughInternal.handleIframe.toString()})()
-      </script>`;
+    return defaultValue;
   }
-
-  return adMarkup;
 }
 
-function handleIframe () {
-  // only load iframe buster JS if we can access the top level document
-  // if we are 'locked in' to this frame then no point trying to bust out: we may as well render in the frame instead
-  var iframeBusterLoaded = false;
-  if (!window.lockedInFrame) {
-    var sfpIframeBusterJs = document.createElement('script');
-    sfpIframeBusterJs.src = 'https://native.sharethrough.com/assets/sfp-set-targeting.js';
-    sfpIframeBusterJs.type = 'text/javascript';
-    try {
-      window.document.getElementsByTagName('body')[0].appendChild(sfpIframeBusterJs);
-      iframeBusterLoaded = true;
-    } catch (e) {
-      utils.logError('Trouble writing frame buster script, error details:', e);
-    }
-  }
-
-  var clientJsLoaded = (!iframeBusterLoaded) ? !!(window.STR && window.STR.Tag) : !!(window.top.STR && window.top.STR.Tag);
-  if (!clientJsLoaded) {
-    var sfpJs = document.createElement('script');
-    sfpJs.src = 'https://native.sharethrough.com/assets/sfp.js';
-    sfpJs.type = 'text/javascript';
-
-    // only add sfp js to window.top if iframe busting successfully loaded; otherwise, add to iframe
-    try {
-      if (iframeBusterLoaded) {
-        window.top.document.getElementsByTagName('body')[0].appendChild(sfpJs);
-      } else {
-        window.document.getElementsByTagName('body')[0].appendChild(sfpJs);
-      }
-    } catch (e) {
-      utils.logError('Trouble writing sfp script, error details:', e);
+function getFloor(bid) {
```

Comment:
Could this be renamed so it does not match the similar function provided by Prebid in the bid?

Reply:
Renamed to `getBidRequestFloor`